### PR TITLE
implement outbound policy

### PR DESCRIPTION
**Pull request by hawkw**

### CODE REVIEW

File: linkerd/service-profiles/src/lib.rs [hunk 1]
```diff
@@ -25,6 +24,7 @@ pub mod http;
 mod proto;
 
 pub use self::client::Client;
+pub use linkerd_client_policy::{LogicalAddr, LookupAddr};
```
Comment (olix0r) [inline linkerd/service-profiles/src/lib.rs:27]: I'd rather keep these things completely decoupled. Let's assume the logicaladdr and lookup types aren't going to be the same.

This PR should ~basically require no changes to anything in service-profiles (Though I think we'll need to cleanup some of the profile/discover stuff as a part of this work).

File: linkerd/client-policy/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    pub addr: Addr,
+}
+
+/// A profile lookup target.
+#[derive(Clone, Hash, Eq, PartialEq)]
+pub struct LookupAddr(pub Addr);
+
+/// A bound logical service address
+#[derive(Clone, Hash, Eq, PartialEq)]
+pub struct LogicalAddr(pub NameAddr);
```
Comment (olix0r) [inline linkerd/client-policy/src/lib.rs:50]: We should be using Meta/ref types for anything related to addrs in policy-land

File: linkerd/client-policy/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    inner: watch::Receiver<Policy>,
+}
+
+// TODO(eliza): this is looking a lot like the `Profile` type...could be made
+// generic eventually...
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Policy {
+    pub addr: Option<LogicalAddr>,
+    pub http_routes: Arc<[http::Route]>,
+    pub opaque_protocol: bool,
```
Comment (olix0r) [inline linkerd/client-policy/src/lib.rs:28]: How does this fit into the policy API in its current form? Where does this configuration come from?

File: linkerd/client-policy/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+use linkerd_proxy_api_resolve as resolve;
+use std::{fmt, net::SocketAddr, str::FromStr, sync::Arc};
+use tokio::sync::watch;
+
+type Backends = Arc<[Backend]>;
+
+#[derive(Clone, Debug)]
+pub struct Receiver {
+    inner: watch::Receiver<Policy>,
+}
```
Comment (olix0r) [inline linkerd/client-policy/src/lib.rs:20]: let's do without this stuff for now and focus on just the `Policy` type for the time being. (I think the profiles::Receiver type isn't so useful in general)

File: linkerd/client-policy/src/lib.rs (hunk 3 above)
Comment (hawkw) [reply linkerd/client-policy/src/lib.rs:28]: there's a proto module converting this...

File: linkerd/client-policy/src/lib.rs (hunk 4 above)
Comment (hawkw) [reply linkerd/client-policy/src/lib.rs:20]: yeah, i thought about getting rid of both of them

File: linkerd/client-policy/src/lib.rs (hunk 2 above)
Comment (hawkw) [reply linkerd/client-policy/src/lib.rs:50]: the logical address is coming from the protobuf's `backend` field; turning that into a meta type will need a change in proxy-api as well.

Comment (olix0r) [reply linkerd/client-policy/src/lib.rs]: Yep. ~Everything about the proxy api draft is on the table to change :) This is why I want to work 'out' from the logical stacks -- let the proxy's needs dictate the API and not vice-versa.

Comment (hawkw) [reply linkerd/client-policy/src/lib.rs]: yeah, that makes sense.

**Pull request closed without merging**